ui/components/component-library/avatar-base/avatar-base.types.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  BackgroundColor,
+  BorderColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+
+export enum AvatarBaseSizes {
+  XS = 'xs',
+  SM = 'sm',
+  MD = 'md',
+  LG = 'lg',
+  XL = 'xl',
+}
+
+export interface AvatarBaseProps {
+  size?: AvatarBaseSizes;
+  children?: React.ReactNode;
+  backgroundColor?: BackgroundColor;
+  borderColor?: BorderColor;
+  color?: TextColor;
+  className?: string;
+}
```

Comment:
Please add the type descriptions from the PropTypes here with appropriate adjustments for typescript